test/Network/MQTT/ParserEncoderSpec.hs

```diff
@@ -0,0 +1,61 @@
+{-# LANGUAGE FlexibleInstances, MultiParamTypeClasses #-}
+
+module Network.MQTT.ParserEncoderSpec       (spec) where
+
+import           Network.MQTT.Packet
+import           Network.MQTT.Encoder       (encodePacket)
+import           Network.MQTT.Parser        (parsePacket)
+
+import           Test.SmallCheck.Series
+import           Test.Hspec
+import           Test.Hspec.SmallCheck      (property)
+
+import           Data.Attoparsec.ByteString (parseOnly)
+import           Data.Word                  (Word16)
+import           Data.ByteString            (ByteString, singleton)
+import           Data.ByteString.Builder    (toLazyByteString)
+import           Data.ByteString.Lazy       (toStrict)
+import qualified Data.Text as T
+
+spec :: Spec
+spec = do
+  describe "Parser-Encoder" $ do
+    it "test identity" $ property $
+      \p -> let encodedPacket = toStrict $ toLazyByteString (encodePacket p)
+                (Right decodedPacket) = parseOnly parsePacket encodedPacket
+            in decodedPacket == (p :: Packet)
+
+instance Monad m => Serial m Word16 where
+  series = generate $ \d -> take (d+1) [ 0..65535 ]
```

Comment:
The usage of depth (`d`) is wrong. That's a depth of the generated recursive structure. While `Word16` is not recursive at all, all values have depth of 1 (or 0, not sure). The proper definition is:
```haskell
series = generate $ cons [minBound .. maxBound]
```

You may reuse definition from `smallcheck-series`, but as far as I see, it prefers low integer values (but that makes little sense for parser/serializer).